daemon/src/taker_cfd.rs

```diff
@@ -116,10 +118,16 @@ impl<O, W> Actor<O, W> {
     }
 
     async fn handle_propose_settlement(&mut self, msg: ProposeSettlement) -> Result<()> {
-        let ProposeSettlement {
-            order_id,
-            current_price,
-        } = msg;
+        let ProposeSettlement { order_id, bid, ask } = msg;
+
+        let mut conn = self.db.acquire().await?;
+
+        #[allow(deprecated)]
+        // We need the position and role to decide the price here. Consider re-modelling access to
+        // quote / where to taker the decision to avoid this call.
+        let cfd = crate::command::load_cfd(order_id, &mut conn).await?;
+
+        let proposal_closing_price = closing_price(bid, ask, cfd.role(), cfd.position());
```

Comment:
I would almost prefer to have two functions for coming up with the closing price. 
- 1. for the projection which needs to know about the position
- 2. for the taker_cfd which doesn't need about the position

We could then enrich the `ProposeSettlement` with what the taker intends to do, e.g. `buy` or `sell` the position. That way we wouldn't need to load from the db here.

Reply:
But I think `taker_cfd` *has* to know about the position! The role is implied there, but the position is not. Loading from the db cannot be optimized away (but we can move it into the `collab_settlement_taker` actor if we want).
The projection has to know both, because it cannot imply either.

I am afraid that by doing this in two places we might run into the problem of changing one place (and not the other) and then having different values. It was a conscious decision to put this decision logic into one place.

Reply:
You are correct, we can only optimize it away if we enrich `ProposeSettlement` which would involve an API change. Any reason why a breaking API change is bad? We always ship the frontend together with the corresponding HTTP Api.

It is a small function which needs 10+ lines of explanation because it is too complex. So I thought it's worth trying to simplify it. Not worth to debate this much longer. :)

Reply:
> Any reason why a breaking API change is bad? We always ship the frontend together with the corresponding HTTP Api.

I don't think there is anything stopping us from changing the API. Might want to think a bit more before doing this. I might give it a follow-up timebox if I find the time. 😬